homeassistant/components/bsblan/config_flow.py

```diff
@@ -41,9 +45,118 @@ async def async_step_user(
         self.username = user_input.get(CONF_USERNAME)
         self.password = user_input.get(CONF_PASSWORD)
 
+        return await self._validate_and_create()
+
+    async def async_step_zeroconf(
+        self, discovery_info: ZeroconfServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle Zeroconf discovery."""
+
+        self.host = str(discovery_info.ip_address)
+        self.port = discovery_info.port or DEFAULT_PORT
+
+        # Try to get MAC from properties first (for new firmware)
+        self.mac = discovery_info.properties.get("mac")
+
+        # If MAC not found in properties, try to parse from raw TXT records
+        if not self.mac and hasattr(discovery_info, "properties_raw"):
+            for key in discovery_info.properties_raw:
+                if key.startswith(b"mac="):
+                    self.mac = key[4:].decode("utf-8")
+                    break
```

Comment:
I kinda agree with what Frenck mentions, `properties_raw` does not exist for this object, where does it come from? the dataclass doesn't have this field

Reply:
This is weird. First it didn't work to retrieve the Mac normally. Maybe it was a fluke? OK I remove the properties_raw as it is indeed not needed.